fix: Resolve path before checking Git bundle

pawamoy opened the pull request:
Closes #791.

I won't be able to add a test right now, but can do later this week.

pawamoy commented:
Ah, the test suite found a breaking test case 🚀 
Fixing it right now.

yajo commented on copier/vcs.py:
```diff
@@ -46,6 +46,7 @@ def is_in_git_repo(path: StrOrPath) -> bool:
 
 def is_git_bundle(path: Path) -> bool:
     """Indicate if a path is a valid git bundle."""
+    path = path.resolve()
     with TemporaryDirectory(prefix=f"{__name__}.is_git_bundle.") as dirname:
         with local.cwd(dirname):
             git("init")
```
You can also do:

```suggestion
    with TemporaryDirectory(prefix=f"{__name__}.is_git_bundle.") as dirname:
        git("-C", dirname, "init")
        return bool(git["-C", dirname, "bundle", "verify", path] & TF)
```

Both solutions are good, so do as you prefer 😄

pawamoy replied:
Oh well, I think I prefer yours: the least directory changes, the better. And that removes the exception on Windows :+1:

yajo reviewed (commented):
I'll set as draft until you get the test stuff. Thanks!

pawamoy commented:
Failure seems unrelated to the change? Also, let me know if you'd like that I add a test for this (not trying to avoid it: I just can't do it from work and will have to do from home).

yajo commented:
Possibly a rebase will fix it.

Please add the test when you have time. No hurry.

pawamoy commented:
Well I found a way to do this from work 😄 

Turns out the `-C` flag changes the current directory as well so I reverted to the `path.resolve()` solution.
Also added a test, and confim it successfully failed without the patch.

pawamoy commented:
Butched up the rebase... that's what happens when I click on buttons instead of running git commands 😩
